core/src/test/java/org/elasticsearch/gateway/MetaDataWriteDataNodesIT.java

```diff
@@ -149,48 +151,52 @@ public void testMetaWrittenWhenIndexIsClosedAndMetaUpdated() throws Exception {
         assertThat(indicesMetaData.get(index).state(), equalTo(IndexMetaData.State.OPEN));
     }
 
-    protected void assertIndexNotInMetaState(String nodeName, String indexName) throws Exception {
-        assertMetaState(nodeName, indexName, false);
+    protected void assertIndexDirectoryDeleted(final String nodeName, final String indexName) throws Exception {
+        assertBusy(new Runnable() {
+                       @Override
+                       public void run() {
+                           logger.info("checking if meta state exists...");
+                           try {
+                               assertFalse("Expecting index directory of " + indexName + " to be deleted from node " + nodeName, indexDirectoryExists(nodeName, indexName));
+                           } catch (Exception e) {
+                               logger.info("failed to check for data director of index {} on node {}", indexName, nodeName);
+                               fail("could not check if data directory still exists");
+                           }
+                       }
+                   }
+        );
     }
 
-    protected void assertIndexInMetaState(String nodeName, String indexName) throws Exception {
-        assertMetaState(nodeName, indexName, true);
+    protected void assertIndexInMetaState(final String nodeName, final String indexName) throws Exception {
+        assertBusy(new Runnable() {
+                       @Override
+                       public void run() {
+                           logger.info("checking if meta state exists...");
+                           try {
+                               assertTrue("Expecting meta state of index " + indexName + " to be on node " + nodeName, getIndicesMetaDataOnNode(nodeName).containsKey(indexName));
+                           } catch (Throwable t) {
+                               logger.info("failed to load meta state", t);
```

Comment:
`getIndicesMetaDataOnNode` throws `Exception` but `run()` in `Runnable` not. Not sure how I can avoid the catch here?